dropwizard-index-page/src/main/java/com/palantir/indexpage/TemplatedAsset.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ */
+
+package com.palantir.indexpage;
+
+import com.google.common.base.Optional;
+import java.util.Map;
+
+/**
+ * Simple interface used by both {@link ClasspathIndexPage} and {@link FileSystemIndexPage} to return the resolved
+ * asset template.
+ */
+public interface TemplatedAsset {
+
+    /**
+     * Returns the content of the asset after resolving all the asset template against the provided {@code context}.
+     */
+    Optional<String> getContent(Map<String, String> context);
```

Comment:
context may be better named something like variables or something related to a template (`templateContext`?)